py-astrovascpy: new BlueBrain Python package

tristan0x opened the pull request.

tristan0x committed:
py-astrovascpy: new BlueBrain Python package

matz-e commented on bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py:
All Python dependencies should be `"build", "run"`, unless they are pure install ([`wheel` is not needed](https://github.com/BlueBrain/spack/blob/develop/lib/spack/spack/build_systems/python.py#L201)).

matz-e commented on bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py:
```diff
@@ -0,0 +1,58 @@
+    depends_on("py-networkx", type="build")
+    depends_on("py-numpy", type="build")
+    depends_on("py-pandas", type="build")
+    depends_on("petsc", type="build")
```
Should just be defaulted type-wise

tristan0x replied:
fixed in [2eda903](https://github.com/BlueBrain/spack/pull/2175/commits/2eda9035d080d19a8b3df2ed08cfb24c27bf3a2a)

tristan0x committed:
py-astrovascpy: fix type of dependencies

* `("build") => ("build", "run")`
* remove py-wheel

matz-e commented on bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py:
```diff
@@ -0,0 +1,57 @@
+    depends_on("py-networkx", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-pandas", type=("build", "run"))
+    depends_on("petsc", type=("build"))
```
```suggestion
    depends_on("petsc")
```
`petsc` does not look like a header-only library.

tristan0x replied:
fixed in [6394ff3](https://github.com/BlueBrain/spack/pull/2175/commits/6394ff3ea1348e15563b27ed93c354f9f96e687a)

tristan0x committed:
py-astrovascpy: remove debug dependency

`PETSC_DIR` env var is now properly set since py-petsc4py is a _run_ dependency

matz-e commented on bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py:
```diff
@@ -0,0 +1,56 @@
+    variant("vtk", default=False, description="add VTK support (mainly for visualization)")
+
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-setuptools-scm", type=("build", "run"))
```
```suggestion
    depends_on("py-setuptools-scm", type="build")
```
I recall this being a build dependency mostly (just saw that when I hit approve :facepalm:)

matz-e replied:
Sorry, that comment may be contradictory to the other one. Pure Python build dependencies should really be `build` only.

tristan0x committed:
Update bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py

Co-authored-by: Matthias Wolf <matthias.wolf@epfl.ch>
